File: app/src/main/java/com/cajetan/youtubeplus/data/VideoDataRepository.java
package com.cajetan.youtubeplus.data;

import android.app.Application;
import android.arch.lifecycle.LiveData;
import android.content.Context;
import android.os.AsyncTask;

import java.util.List;

public class VideoDataRepository {
    private VideoDataDao mVideoDataDao;
    private LiveData<List<VideoData>> mAllVideoData;

    VideoDataRepository(Application application) {
        VideoDatabase db = VideoDatabase.getDatabase(application);
        mVideoDataDao = db.videoDataDao();
        mAllVideoData = mVideoDataDao.getAll();
    }

    LiveData<List<VideoData>> getAllVideoData() {
        return mAllVideoData;
    }

    void insert(VideoData videoData) {
        new InsertTask(mVideoDataDao).execute(videoData);
    }

    void delete(VideoData videoData) {
        new DeleteTask(mVideoDataDao).execute(videoData);
    }

    private static class InsertTask extends AsyncTask<VideoData, Void, Void> {
        private VideoDataDao mAsyncTaskDao;

        InsertTask(VideoDataDao dao) {
            mAsyncTaskDao = dao;
        }

        @Override
        protected Void doInBackground(VideoData... videoData) {
            mAsyncTaskDao.insertAll(videoData[0]);
            return null;
        }
    }

    private static class DeleteTask extends AsyncTask<VideoData, Void, Void> {
        private VideoDataDao mAsyncTaskDao;

        DeleteTask(VideoDataDao dao) {
            mAsyncTaskDao = dao;
        }

        @Override
        protected Void doInBackground(VideoData... videoData) {
            mAsyncTaskDao.delete(videoData[0]);
            return null;
        }
    }
}

File: app/src/main/java/com/cajetan/youtubeplus/data/VideoDataRepository.kt
package com.cajetan.youtubeplus.data

import android.app.Application
import android.util.Log
import androidx.lifecycle.LiveData
import org.jetbrains.anko.doAsync

class VideoDataRepository(application: Application) {
    private var mVideoDataDao: VideoDataDao = VideoDatabase.getDatabase(application)!!.videoDataDao()
    private var mAllVideoData: LiveData<List<VideoData>>

    init {
        mAllVideoData = mVideoDataDao.getAll()
    }

    fun getAllVideoData(): LiveData<List<VideoData>> {
        return mAllVideoData
    }

    fun contains(videoId: String): Boolean {
        Log.d("Repository", "Size with $videoId is ${mVideoDataDao.getWithId(videoId)}")
        return mVideoDataDao.getWithId(videoId) != null
    }

    fun insert(videoData: VideoData) {
        doAsync {
            mVideoDataDao.insertAll(videoData)
        }
    }

    fun delete(videoData: VideoData) {
        doAsync {
            mVideoDataDao.delete(videoData)
        }
    }
}